nullaway/src/main/java/com/uber/nullaway/handlers/CompositeHandler.java

```diff
@@ -184,7 +184,10 @@ public NullnessHint onDataflowVisitMethodInvocation(
       NullnessHint n =
           h.onDataflowVisitMethodInvocation(
               node, state, apContext, inputs, thenUpdates, elseUpdates, bothUpdates);
-      nullnessHint = nullnessHint.merge(n);
+      nullnessHint = nullnessHint.mergeByPriority(n);
+      if (nullnessHint.isHighestPriority()) {
```

Comment:
Not exactly related to this PR, but in general, I feel like our use of `CompositeHandler` is set up for a potentially-bad future bug where multiple handlers implement a particular method and their interaction is unexpected or order-dependent.  @lazaroclapp do you know if we have known cases where multiple handlers try to "handle" the same construct and we expect one to win?  If not, we may want to implement some kind of "handler debug mode" (off by default) where we actually check in `CompositeHandler` that only one contained handler is returning a non-default value for each case.  It will make the code messier but may save us some pain in the future.

Reply:
This would not work:

> check in `CompositeHandler` that only one contained handler is returning a non-default value for each case

Handlers are meant to override and use the computations of one another, which is why they execute in a chain with meaningful order between them. Each `Handler` API method defines how results from different handlers are combined, in the case of `onDataflowVisitMethodInvocation` (admittedly one of the most complex) it's based on the priority given by the `NullnessHint` enum. This definitely leads to weird corner cases, often, but it also gives us a lot of flexibility in deciding which "source" should win for any given nullness fact (e.g. between jar annotations and library models).